### ISSUE TITLE
'Pay Now' in My Telstra app not working

### ISSUE THREAD

**Issue by gyCfjSnO** [hagezi/dns-blocklists#697]
When using HaGeZi Personal Black & White blocklist, the 'Pay Now' feature in the [My Telstra app](https://www.telstra.com.au/mytelstra) does not work and I cannot pay my mobile bill.

**Comment by hagezi** [comment]
Which blocked domain leads to this behavior?

**Comment by gyCfjSnO** [comment]
Sorry didn't realise you'd respond so quickly. I have a list of almost 15 domains I'm trying to narrow down. I'll post a smaller list shortly, having great difficulty to pinpoint the domain. When I can get it working another issue pops up regarding the secure pay with Visa option not loading and also needing to be unblocked.

**Comment by hagezi** [comment]
Can anyone help, can not test this myself?

**Comment by gyCfjSnO** [comment]
So far I have had to unblock these to get the 'Pay Now' screen to work.
```
@@||ap1.thunderhead.com^$important
@@||telstra.demdex.net^$important
@@||ap1.cdn.thunderhead.com^$important
@@||telstracorporationlt.tt.omtrdc.net^$important
@@||cdn.branch.io^$important
@@||infos.telstra.com.au^$important
@@||cm.everesttech.net^$important
@@||fls.doubleclick.net^$important
@@||mobile-collector.newrelic.com^$important
@@||app-measurement.com^$important
@@||launches.appsflyer.com^$important
@@||mcxlfzpz83f0lpzmy3wy628h7zh1.device.marketingcloudapis.com^$important
@@||cdn.betrad.com^$important
@@||mboxedge36.tt.omtrdc.net^$important
```
And I have had to unblock these to get the '[Click to Pay](https://www.mastercard.com.au/en-au/personal/ways-to-pay/click-to-pay.html)' screen to work, inside of the 'Pay Now' screen.
```
@@||googleads.g.doubleclick.net^$important
@@||stats.g.doubleclick.net^$important
@@||stun.cdn-net.com^$important
@@||errors.client.optimizely.com^$important
```

**Comment by hagezi** [comment]
WTF

**Comment by gyCfjSnO** [comment]
I tested on two different mobiles as well.

The app works flawlessly when AdGuard protection or the list is disabled.

**Comment by hagezi** [comment]
If these are really all needed, then this will not work with any blocklist. Lots of popular trackers and ad domains. I would ask the Hersxteller what that is supposed to ...

**Comment by hagezi** [comment]
I think that the following cause the problem:

```
infos.telstra.com.au
mastercard.com.ssl.sc.omtrdc.net
mboxedge36.tt.omtrdc.net
smetrics.mastercard.com
telstra.com.au.ssl.sc.omtrdc.net
telstracorporationlt.tt.omtrdc.net
```

**Comment by gyCfjSnO** [comment]
> @gyCfjSnO new Version is online, please test without your whitelisting.

No luck unfortunately, I then tried manually whitelisting ```ap1.thunderhead.com``` & ```ap1.cdn.thunderhead.com``` which didn't work BUT then the domains ```cdn.branch.io``` & ```cm.everesttech.net``` were both being queried. I didn't want to start whitelisting domains again as I'd basically end up back at square one where I have a massive list...

**Comment by hagezi** [comment]
Does it work if you use the AdguardDNS list alone?

**Comment by gyCfjSnO** [comment]
> Does it work if you use the AdguardDNS list alone?

I updated my reply about btw.

**Comment by gyCfjSnO** [comment]
> Does it work if you use the AdguardDNS list alone?

No it doesn't load when using online AdGuard list.

**Comment by hagezi** [comment]
OISD only?

**Comment by gyCfjSnO** [comment]
> OISD only?

Using the default OISD Basic list I get a glimpse of the 'Something went wrong' screen but then it quickly reloads the screen and it works, however the PayPal and Click to Pay buttons don't work.

**Comment by hagezi** [comment]
OK, then I guess you'll have to disable AdBlocker for the payment process. The app seems to be the absolute plague ...
